_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,206 @@
+---
+name: iframe has a unique accessible name
+
+description: |
+ The accessible name for each iframe is unique
+
+success_criterion:
+- 4.1.2
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to `iframe` elements that are [exposed to assistive technologies](#exposed-to-assistive-technologies) and have a [non-empty](#non-empty) [accessible name](#accessible-name).
+
+### Expectation
+
+Each target element has an [accessible name](#accessible-name) that is unique across all other accessible names for `iframes` within the same [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees) as the target element.
+
+**Note:** 
+- `iframes` with the same `src` can have the same [accessible name](#accessible-name).
+- Nested `iframe` contents treat the parent `iframe` element as the [document tree](https://www.w3.org/TR/dom41/#document-trees).
```

Comment:
I don't think this is perfectly correct, or else I just don't understand it well. 
May I suggest a clean up of the terms, e.g.: 

"An `iframe` creates an embedded browsing context. The content of this browsing context is not part of the [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees) that it is embedded in."